janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/JointIndexQuery.java

```diff
@@ -100,9 +100,11 @@ public String toString() {
 
     @Override
     public JointIndexQuery updateLimit(int newLimit) {
-        JointIndexQuery ji = new JointIndexQuery(Lists.newArrayList(queries));
-        ji.setLimit(newLimit);
-        return ji;
+        List<Subquery> subqueries = (queries.size() == 1) ? Lists.newArrayList(queries.get(0).updateLimit(newLimit)) :
+            Lists.newArrayList(queries);
+        JointIndexQuery jointIndexQuery = new JointIndexQuery(subqueries);
+        jointIndexQuery.setLimit(newLimit);
+        return new JointIndexQuery(subqueries);
```

Comment:
I am confused here.
Why do we create `jointIndexQuery` and then don't return it? Why do we need to return `new JointIndexQuery(subqueries)`?

Reply:
Yes, this is typo